src/main/java/org/opensrp/service/EventService.java

```diff
@@ -189,6 +203,16 @@ public synchronized Event addEvent(Event event, String username) {
 					"An event already exists with given baseEntity and formSubmission combination. Consider updating");
 		}
 
+		if (event.getDetails() != null
+				&& StringUtils.isNotBlank(event.getDetails().get(CASE_NUMBER))){
```

Comment:
First check that eventType = `Case_Details` so as to avoid making the follow up checks for other types of events